pkg/kncloudevents/retries.go

```diff
@@ -83,7 +83,7 @@ func RetryConfigFromDeliverySpec(spec v1.DeliverySpec) (RetryConfig, error) {
 
 	retryConfig := NoRetries()
 
-	retryConfig.CheckRetry = RetryIfGreaterThan300
+	retryConfig.CheckRetry = SelectiveRetry
```

Comment:
this is a behavioral change, while I agree that the `SelectiveRetry` is more compliant to the spec
Any rekt/e2e tests ?

Reply:
Thank you for your kind comment. I also agree that it is behavior changes.

I will try to add rekt or e2e tests! But, I'm not familiar with those tests... 

So, Am I right to refer to `test/rekt/...` and `test/e2e/...`?

Reply:
@WoWsj for e2e tests you can take a look at expanding this test [1], with something like:

```golang
diff --git a/test/rekt/features/channel/data_plane.go b/test/rekt/features/channel/data_plane.go
index 76a4967fd..0d8fef863 100644
--- a/test/rekt/features/channel/data_plane.go
+++ b/test/rekt/features/channel/data_plane.go
@@ -178,6 +178,21 @@ func addControlPlaneDelivery(fs *feature.FeatureSet) {
 				Retry:          ptr.Int32(1),
 			},
 		}},
+	}, {
+		name: "Channels SHOULD NOT retry resending CloudEvents when subscriber responds with 400",
+		subs: []subCfg{{
+			prefix:            "sub1",
+			hasSub:            true,
+			subFailCount:      1,
+			subFailStatusCode: 400,
+			subReplies:        false,
+			hasReply:          false,
+			replyFailCount:    0,
+			delivery: &v1.DeliverySpec{
+				DeadLetterSink: new(duckv1.Destination),
+				Retry:          ptr.Int32(1),
+			},
+		}},
 	}, {
 		name: "Channels SHOULD support various retry configuration parameters: [the maximum number of retries]",
 		subs: []subCfg{{
diff --git a/test/rekt/features/channel/topology.go b/test/rekt/features/channel/topology.go
index a44514fe8..95d552b1b 100644
--- a/test/rekt/features/channel/topology.go
+++ b/test/rekt/features/channel/topology.go
@@ -20,24 +20,26 @@ import (
 	"fmt"
 	"strconv"
 
+	"knative.dev/reconciler-test/pkg/eventshub"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/pkg/manifest"
+
 	v1 "knative.dev/eventing/pkg/apis/duck/v1"
 	"knative.dev/eventing/test/rekt/features/knconf"
 	"knative.dev/eventing/test/rekt/resources/channel_impl"
 	"knative.dev/eventing/test/rekt/resources/delivery"
 	"knative.dev/eventing/test/rekt/resources/subscription"
-	"knative.dev/reconciler-test/pkg/eventshub"
-	"knative.dev/reconciler-test/pkg/feature"
-	"knative.dev/reconciler-test/pkg/manifest"
 )
 
 type subCfg struct {
-	prefix         string
-	hasSub         bool
-	subFailCount   uint
-	subReplies     bool
-	hasReply       bool
-	replyFailCount uint
-	delivery       *v1.DeliverySpec
+	prefix            string
+	hasSub            bool
+	subFailCount      uint
+	subReplies        bool
+	hasReply          bool
+	replyFailCount    uint
+	delivery          *v1.DeliverySpec
+	subFailStatusCode int
 }
 
 func (s *subCfg) subName() string {
@@ -92,6 +94,9 @@ func createChannelTopology(f *feature.Feature, chName string, chDS *v1.DeliveryS
 		// Install the expected sinks, they all might not be used.
 
 		subOpts := []eventshub.EventsHubOption{eventshub.DropFirstN(sub.subFailCount)}
+		if sub.subFailStatusCode > 0 {
+			subOpts = append(subOpts, eventshub.DropEventsResponseCode(sub.subFailStatusCode))
+		}
 		if sub.subReplies {
 			subOpts = append(subOpts, eventshub.ReplyWithAppendedData(sub.prefix))
 		}

```

[1] https://github.com/knative/eventing/blob/7e899fd166deb40c885d3fc4fb9e2901375a7bff/test/rekt/features/channel/data_plane.go#L157

Reply:
@pierDipi I tried to run `rekt` test, but failed. Is it possible to run locally? 😭 

When i run this:

```
SYSTEM_NAMESPACE=knative-eventing go test -count=1 -v -tags=e2e ./test/rekt/...
```

It created a pod, and its status returned `InvalidImageName` like this

```
$ kgp
NAMESPACE          NAME                                  READY   STATUS             RESTARTS       AGE
test-rlhvpujz      dls-ohdblmic                          0/1     InvalidImageName   0              60s
```

its manifest:

```
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2023-01-23T08:01:28Z"
  labels:
    app: eventshub-dls-ohdblmic
  name: dls-ohdblmic
  namespace: test-rlhvpujz
  resourceVersion: "20656"
  uid: 83bb35ee-481c-4727-8311-0f1ba67ab31b
spec:
  containers:
  - env:
    - name: SYSTEM_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    - name: POD_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.name
    - name: EVENT_LOGS
      value: recorder,logger
    - name: EVENT_GENERATORS
      value: receiver
    - name: K_CONFIG_LOGGING
      value: '{"zap-logger-config":"{\n  \"level\": \"info\",\n  \"development\":
        false,\n  \"outputPaths\": [\"stdout\"],\n  \"errorOutputPaths\": [\"stderr\"],\n  \"encoding\":
        \"json\",\n  \"encoderConfig\": {\n    \"timeKey\": \"ts\",\n    \"levelKey\":
        \"level\",\n    \"nameKey\": \"logger\",\n    \"callerKey\": \"caller\",\n    \"messageKey\":
        \"msg\",\n    \"stacktraceKey\": \"stacktrace\",\n    \"lineEnding\": \"\",\n    \"levelEncoder\":
        \"\",\n    \"timeEncoder\": \"iso8601\",\n    \"durationEncoder\": \"\",\n    \"callerEncoder\":
        \"\"\n  }\n}\n"}'
    - name: K_CONFIG_TRACING
      value: '{"backend":"none","debug":"false","sample-rate":"0.1"}'
    image: ko://knative.dev/reconciler-test/cmd/eventshub
    imagePullPolicy: IfNotPresent
    name: eventshub
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /health/ready
        port: 8080
        scheme: HTTP
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-lrwzl
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: minikube
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: test-rlhvpujz
  serviceAccountName: test-rlhvpujz
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: kube-api-access-lrwzl
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
```

`image: ko://knative.dev/reconciler-test/cmd/eventshub` 🤔 Why Ko can't replace its image URL...?

Reply:
@pierDipi Could I ask about this issue? I have no clue... 🤔

Reply:
Can you paste the logs?

You should see something like
```
logger.go:130: 2023-01-20T05:53:02.141Z	DEBUG	ko/cmd.go:70	google/ko publish: ["go" "run" "github.com/google/ko@v0.11.2" "publish" "-B" "knative.dev/reconciler-test/cmd/eventshub"]	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.443Z	DEBUG	ko/cmd.go:55	go: downloading github.com/google/go-containerregistry v0.8.1-0.20220209165246-a44adc326839	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.443Z	DEBUG	ko/cmd.go:55	go: downloading github.com/chrismellard/docker-credential-acr-env v0.0.0-20220119192733-fe33c00cee21	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.443Z	DEBUG	ko/cmd.go:55	go: downloading github.com/awslabs/amazon-ecr-credential-helper/ecr-login v0.0.0-20220216180153-3d7835abdf40	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.443Z	DEBUG	ko/cmd.go:55	go: downloading github.com/spf13/cobra v1.4.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.443Z	DEBUG	ko/cmd.go:55	go: downloading go.uber.org/automaxprocs v1.4.1-0.20220314153950-975e177ad84f	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.444Z	DEBUG	ko/cmd.go:55	go: downloading golang.org/x/sync v0.0.0-20210220032951-036812b2e83c	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.444Z	DEBUG	ko/cmd.go:55	go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.455Z	DEBUG	ko/cmd.go:55	go: downloading k8s.io/apimachinery v0.23.5	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.465Z	DEBUG	ko/cmd.go:55	go: downloading github.com/containerd/stargz-snapshotter/estargz v0.11.3	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.473Z	DEBUG	ko/cmd.go:55	go: downloading github.com/opencontainers/image-spec v1.0.3-0.20220114050600-8b9d41f48198	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.495Z	DEBUG	ko/cmd.go:55	go: downloading github.com/sigstore/cosign v1.3.2-0.20211120003522-90e2dcfe7b92	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.624Z	DEBUG	ko/cmd.go:55	go: downloading golang.org/x/tools v0.1.10	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.624Z	DEBUG	ko/cmd.go:55	go: downloading github.com/spf13/viper v1.10.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.624Z	DEBUG	ko/cmd.go:55	go: downloading github.com/dprotaso/go-yit v0.0.0-20191028211022-135eb7262960	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.625Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest/autorest/azure/auth v0.5.8	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.625Z	DEBUG	ko/cmd.go:55	go: downloading github.com/docker/docker-credential-helpers v0.6.4	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.626Z	DEBUG	ko/cmd.go:55	go: downloading github.com/klauspost/compress v1.15.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.626Z	DEBUG	ko/cmd.go:55	go: downloading github.com/opencontainers/go-digest v1.0.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.626Z	DEBUG	ko/cmd.go:55	go: downloading github.com/vbatts/tar-split v0.11.2	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.632Z	DEBUG	ko/cmd.go:55	go: downloading github.com/google/go-cmp v0.5.7	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.633Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest/autorest v0.11.19	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.633Z	DEBUG	ko/cmd.go:55	go: downloading k8s.io/klog/v2 v2.30.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.637Z	DEBUG	ko/cmd.go:55	go: downloading github.com/docker/docker v20.10.13+incompatible	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.690Z	DEBUG	ko/cmd.go:55	go: downloading k8s.io/klog v1.0.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.712Z	DEBUG	ko/cmd.go:55	go: downloading github.com/docker/cli v20.10.12+incompatible	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.713Z	DEBUG	ko/cmd.go:55	go: downloading github.com/mitchellh/go-homedir v1.1.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.715Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest v14.2.0+incompatible	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.718Z	DEBUG	ko/cmd.go:55	go: downloading sigs.k8s.io/kind v0.12.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.951Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/azure-sdk-for-go v55.8.0+incompatible	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.951Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest/autorest/adal v0.9.13	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.960Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2 v1.13.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.961Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/config v1.13.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.961Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/service/ecr v1.14.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.961Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/service/ecrpublic v1.11.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.961Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/smithy-go v1.10.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.964Z	DEBUG	ko/cmd.go:55	go: downloading k8s.io/utils v0.0.0-20211116205334-6203023598ed	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:05.964Z	DEBUG	ko/cmd.go:55	go: downloading github.com/go-logr/logr v1.2.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.044Z	DEBUG	ko/cmd.go:55	go: downloading github.com/fsnotify/fsnotify v1.5.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.052Z	DEBUG	ko/cmd.go:55	go: downloading github.com/magiconair/properties v1.8.5	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.061Z	DEBUG	ko/cmd.go:55	go: downloading github.com/mitchellh/mapstructure v1.4.3	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.080Z	DEBUG	ko/cmd.go:55	go: downloading github.com/spf13/afero v1.6.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.085Z	DEBUG	ko/cmd.go:55	go: downloading github.com/spf13/cast v1.4.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.153Z	DEBUG	ko/cmd.go:55	go: downloading github.com/spf13/jwalterweatherman v1.1.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.153Z	DEBUG	ko/cmd.go:55	go: downloading github.com/subosito/gotenv v1.2.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.156Z	DEBUG	ko/cmd.go:55	go: downloading gopkg.in/ini.v1 v1.66.2	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.165Z	DEBUG	ko/cmd.go:55	go: downloading github.com/docker/distribution v2.7.1+incompatible	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.174Z	DEBUG	ko/cmd.go:55	go: downloading cloud.google.com/go v0.99.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.251Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest/autorest/azure/cli v0.4.2	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.251Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest/logger v0.2.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.255Z	DEBUG	ko/cmd.go:55	go: downloading github.com/dimchansky/utfbom v1.1.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.263Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest/tracing v0.6.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.264Z	DEBUG	ko/cmd.go:55	go: downloading github.com/sigstore/sigstore v1.0.2-0.20211115214857-534e133ebf9d	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.267Z	DEBUG	ko/cmd.go:55	go: downloading github.com/Azure/go-autorest/autorest/date v0.3.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.272Z	DEBUG	ko/cmd.go:55	go: downloading github.com/form3tech-oss/jwt-go v3.2.5+incompatible	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.345Z	DEBUG	ko/cmd.go:55	go: downloading golang.org/x/crypto v0.0.0-20220112180741-5e0467b6c7ce	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.347Z	DEBUG	ko/cmd.go:55	go: downloading github.com/hashicorp/hcl v1.0.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.347Z	DEBUG	ko/cmd.go:55	go: downloading github.com/pelletier/go-toml v1.9.4	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.374Z	DEBUG	ko/cmd.go:55	go: downloading golang.org/x/net v0.0.0-20220127074510-2fabfed7e28f	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.394Z	DEBUG	ko/cmd.go:55	go: downloading github.com/alessio/shellescape v1.4.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.455Z	DEBUG	ko/cmd.go:55	go: downloading github.com/theupdateframework/go-tuf v0.0.0-20210722233521-90e262754396	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.466Z	DEBUG	ko/cmd.go:55	go: downloading golang.org/x/term v0.0.0-20210927222741-03fcf44c2211	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.466Z	DEBUG	ko/cmd.go:55	go: downloading sigs.k8s.io/yaml v1.3.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.472Z	DEBUG	ko/cmd.go:55	go: downloading github.com/BurntSushi/toml v0.4.1	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.472Z	DEBUG	ko/cmd.go:55	go: downloading github.com/evanphx/json-patch/v5 v5.6.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.479Z	DEBUG	ko/cmd.go:55	go: downloading github.com/evanphx/json-patch v4.12.0+incompatible	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.545Z	DEBUG	ko/cmd.go:55	go: downloading github.com/mattn/go-isatty v0.0.14	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:06.881Z	DEBUG	ko/cmd.go:55	go: downloading golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.008Z	DEBUG	ko/cmd.go:55	go: downloading github.com/docker/go-units v0.4.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.008Z	DEBUG	ko/cmd.go:55	go: downloading github.com/docker/go-connections v0.4.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.008Z	DEBUG	ko/cmd.go:55	go: downloading github.com/gogo/protobuf v1.3.2	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.009Z	DEBUG	ko/cmd.go:55	go: downloading github.com/containerd/containerd v1.5.9	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.009Z	DEBUG	ko/cmd.go:55	go: downloading google.golang.org/genproto v0.0.0-20220126215142-9970aeb2e350	{"test": "TestBrokerWithDLQ"}
=== CONT  TestBrokerWithDLQ
    logger.go:130: 2023-01-20T05:53:07.452Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/internal/configsources v1.1.4	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.453Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.10.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.453Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/internal/ini v1.3.5	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.453Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/credentials v1.8.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.453Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/service/sso v1.9.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.453Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/service/sts v1.14.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.453Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/internal/endpoints/v2 v2.2.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.453Z	DEBUG	ko/cmd.go:55	go: downloading github.com/jmespath/go-jmespath v0.4.0	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:07.472Z	DEBUG	ko/cmd.go:55	go: downloading github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.7.0	{"test": "TestBrokerWithDLQ"}
=== CONT  TestBrokerWithDLQ
    logger.go:130: 2023-01-20T05:53:28.523Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:28 Using base gcr.io/distroless/static:nonroot@sha256:1fa522fe6cfe020d50341f1ca561c099487bd44f8eb98d25d1920b07e05e40be for knative.dev/reconciler-test/cmd/eventshub	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:29.071Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:29 Building knative.dev/reconciler-test/cmd/eventshub for linux/amd64	{"test": "TestBrokerWithDLQ"}
=== CONT  TestBrokerWithDLQ
    logger.go:130: 2023-01-20T05:53:34.364Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:34 Publishing gcr.io/knative-boskos-37/eventing-e2e-img/14065/eventshub:latest	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:35.286Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:35 pushed blob: sha256:90d246dc1eefc0ce646f0e190dd19f8f45d4f4f8c612bd9e2f10aa710f25b925	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:35.319Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:35 pushed blob: sha256:51a1470c5a23be7930f95a8945cc89f9fe1c29ac0826f3a86234030b96b798d9	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:35.524Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:35 gcr.io/knative-boskos-37/eventing-e2e-img/14065/eventshub:sha256-39f64a836021a45f252d789274e31410b9a54e724c85b425b90c239514f6de5a.sbom: digest: sha256:bac36c95ddb6e0aee62e5318545fd001125fe871c30eacf1b236566d4ee7fc2f size: 369	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:35.524Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:35 Published SBOM gcr.io/knative-boskos-37/eventing-e2e-img/14065/eventshub:sha256-39f64a836021a45f252d789274e31410b9a54e724c85b425b90c239514f6de5a.sbom	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:35.625Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:35 existing blob: sha256:fc251a6e798157dc3b46fd265da72f39cd848e3f9f4a0b28587d1713b878deb9	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:35.625Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:35 existing blob: sha256:250c06f7c38e52dc77e5c7586c3e40280dc7ff9bb9007c396e06d96736cf8542	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:36.441Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:36 pushed blob: sha256:8099d8f775b8a0bbf62a523e74253cc95e9e74789b7805e1459f51874b1d5966	{"test": "TestBrokerWithDLQ"}
=== CONT  TestBrokerWithDLQ
    logger.go:130: 2023-01-20T05:53:37.820Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:37 pushed blob: sha256:c9c205be8c97a75958659f373633a8bef2219078deae0d0d65394fb3d999cb55	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:38.055Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:38 gcr.io/knative-boskos-37/eventing-e2e-img/14065/eventshub:latest: digest: sha256:39f64a836021a45f252d789274e31410b9a54e724c85b425b90c239514f6de5a size: 752	{"test": "TestBrokerWithDLQ"}
    logger.go:130: 2023-01-20T05:53:38.056Z	DEBUG	ko/cmd.go:55	2023/01/20 05:53:38 Published gcr.io/knative-boskos-37/eventing-e2e-img/14065/eventshub@sha256:39f64a836021a45f252d789274e31410b9a54e724c85b425b90c239514f6de5a	{"test": "TestBrokerWithDLQ"}
```